src/olympia/blocklist/models.py

```diff
@@ -83,20 +83,19 @@ def addon_versions(self):
 
     @classmethod
     def preload_addon_versions(cls, blocks):
-        """Preload block.addon_versions into a list of blocks.
-        If you're calling this on a list of blocks it's expected that you've
-        set cached_property self.addon in a db efficient way beforehand.
-        """
-        addon_ids = [block.addon.id for block in blocks if block.addon]
-        qs = Version.unfiltered.filter(addon_id__in=addon_ids).order_by(
-            'id').values('addon_id', 'version', 'id', 'channel')
-        addons_versions = defaultdict(OrderedDict)
+        """Preload block.addon_versions into a list of blocks."""
+        block_guids = [block.guid for block in blocks]
+        GUID = 'addon__guid'
+        qs = (
+            Version.unfiltered.filter(**{f'{GUID}__in': block_guids})
+                              .order_by('id')
+                              .annotate(**{GUID: models.F(GUID)}))
```

Comment:
Do we want the default transformers to be called ? That's a few extra queries to fetch translations, files, applicationversions, and then build the related model instances.

Reply:
https://github.com/mozilla/addons-server/pull/14453/commits/2407bf7cbd0bb62f6a0335f02e5b9aa676c39ffd